packages/opentelemetry-core/src/trace/TraceState.ts

```diff
@@ -65,9 +63,9 @@ export class TraceState implements types.TraceState {
 
   private _parse(rawTraceState: string) {
     if (rawTraceState.length > MAX_TRACE_STATE_LEN) return;
-    // TODO validate maximum number of items
     this._internalState = rawTraceState
       .split(LIST_MEMBERS_SEPARATOR)
+      .slice(0, MAX_TRACE_STATE_ITEMS)
```

Comment:
It is possible that we would ignore some valid states using this approach. I think we should do this when we validated all the states or during validation process.

Reply:
Should invalid state items not count towards the maximum limit (since we are just discarding that item instead of the entire tracestate for this tracestate impl)? If so then I agree, it would make sense to move this to after the validation reducer.

Reply:
> Should invalid state items not count towards the maximum limit?

As we are gonna ignore invalid state items down below, IMO it should not count towards the max limit.

Reply:
I've moved the truncation to be after all of the validation is performed (99e4cc6). If using a `List` instead of a `Map` for the `_internalState` structure turns out to be more performant, I think truncation would be able to be made simpler